nixos/modules/programs/rootston.nix

```diff
@@ -53,32 +49,50 @@ in {
     };
 
     config = mkOption {
-      type = types.str;
-      default = ''
-        [keyboard]
-        meta-key = Logo
+      type = types.attrs;
+      default = {
+        keyboard = {
+          meta-key = "Logo";
+        };
 
-        # Sway/i3 like Keybindings
-        # Maps key combinations with commands to execute
-        # Commands include:
-        # - "exit" to stop the compositor
-        # - "exec" to execute a shell command
-        # - "close" to close the current view
-        # - "next_window" to cycle through windows
-        [bindings]
-        Logo+Shift+e = exit
-        Logo+q = close
-        Logo+m = maximize
-        Alt+Tab = next_window
-        Logo+Return = exec weston-terminal
-        Logo+d = exec rofi -show run
-      '';
+        bindings = {
+          # Sway/i3 like Keybindings
+          # Maps key combinations with commands to execute
+          # Commands include:
+          # - "exit" to stop the compositor
+          # - "exec" to execute a shell command
+          # - "close" to close the current view
+          # - "next_window" to cycle through windows
+          "Logo+Shift+e" = "exit";
+          "Logo+q" = "close";
+          "Logo+m" = "maximize";
+          "Alt+Tab" = "next_window";
+          "Logo+Return" = "exec weston-terminal";
+          "Logo+d" = "exec rofi -show run";
+        };
+      };
       description = ''
         Default configuration for rootston (used when called without any
         parameters).
       '';
     };
 
+    extraConfig = mkOption {
+      type = types.attrs;
```

Comment:
`toINI` only accepts `attrsets of attrsets of any`, anything else will make it crash iirc. So this should really be `attrsOf (attrsOf undefined)`.

Reply:
Or even better would be an `ini` type that only accepts stuff `toINI` also accepts, but I haven’t thought that through.

Reply:
@Profpatsch Thank you very much for your review and apologies that I had to close this PR.